integrated_channels/moodle/client.py

```diff
@@ -311,38 +312,191 @@ def _wrapped_create_course_completion(self, user_id, payload):
         return self._post(params)
 
     @moodle_request_wrapper
-    def create_content_metadata(self, serialized_data):
+    def _wrapped_create_content_metadata(self, payload):
+        return self._post(payload)
+
+    @moodle_request_wrapper
+    def _wrapped_update_content_metadata(self, payload):
+        return self._post(payload)
+
+    @moodle_request_wrapper
+    def _get_course_forum(self, course_id):
         """
-        The below assumes the data is dict/object.
-        Format should look like:
-        {
-          courses[0][shortname]: 'value',
-          courses[0][fullname]: 'value',
-          [...]
-          courses[1][shortname]: 'value',
-          courses[1][fullname]: 'value',
-          [...]
+        Obtains a list of forums for a given course id.
+        """
+        params = {
+            'wsfunction': 'mod_forum_get_forums_by_courses',
+            'courseids[]': course_id,
         }
+        return self._post(params)
+
+    def get_announcement_forum(self, course_id):
         """
-        serialized_data['wsfunction'] = 'core_course_create_courses'
-        response = self._post(serialized_data)
-        return response
+        The response is a list type, because of course it is.
+        For new courses, there should only be 1 forum and it should be Announcements
+        """
+        response = self._get_course_forum(course_id)
+        return response[0]['id']
+
+    @moodle_request_wrapper
+    def _get_forum_discussions(self, forum_id):
+        """
+        Returns discussion posts found in the specified forum
+        """
+        params = {
+            'wsfunction': 'mod_forum_get_forum_discussions',
+            'forumid': forum_id,
+            'moodlewsrestformat': 'json',
+            'wstoken': self.token,
+        }
+        return self._post(params)
+
+    def get_announcement_post(self, forum_id):
```

Comment:
yeah I really hope they come up with a new api version that is actually higher up in the maturity model so we can actually read and use it

Reply:
I mean moodle isn't the one using "announcement" anywhere. I specify that since there COULD (in a very theoretical sense that would likely break our integration) be other forums and definitely other posts.

When you limit to single activity, the one forum is added to the one activity which at creation time is the Announcements section. That's why I specified announcement in the names.

Reply:
After investigation, if we specify single activity or social formats, there can be only one ™️ 
But there can be other discussions and posts in said forum. What is the difference between a discussion and post? Discussion is like a reddit post and posts are exclusively replies... Hopefully.

Reply:
>  Discussion is like a reddit post and posts are exclusively replies... Hopefully.

This actually helps a lot with my understanding 👍. If only **their** wording could be more clear...